### PR TITLE
[GDNative] merge of NativeScript and GDNative, new GDNative singletons

### PR DESCRIPTION
**Pull request by karroffel**
A GDNativeLibrary now has a field "singleton_gdnative" which can be
used to let the `godot_gdnative_singleton` procedure be executed on
Godot's startup. In future this can be used to register new
scripting languages or resource importer types.

### CODE REVIEW

File: modules/gdnative/nativescript/nativescript.h [hunk 1]
```diff
@@ -38,8 +38,8 @@
 #include "script_language.h"
 #include "self_list.h"
 
-#include "godot_nativescript.h"
 #include "modules/gdnative/gdnative.h"
+#include <nativescript/godot_nativescript.h>
```
Comment (akien-mga) [inline modules/gdnative/nativescript/nativescript.h]: Use `""` for Godot-specific includes, this is not a system wide one.

Comment (karroffel) [reply modules/gdnative/nativescript/nativescript.h]: It is. It's in the `CPPPATH`

Comment (akien-mga) [reply modules/gdnative/nativescript/nativescript.h]: Same comment for all other `#include <.*>` you're using.

Comment (karroffel) [reply modules/gdnative/nativescript/nativescript.h]: To explain a little bit further:

The GDNative module is now split into an implementation part and an interface part. The interface part is all inside `modules/gdnative/include`. This makes it easier to extract those files so libraries can use them.

That's why I added that path into the include path for that module.

Comment (akien-mga) [reply modules/gdnative/nativescript/nativescript.h]: Still, our convention is to use `""` for local headers, `<>` for system ones. Being in the CPPPATH doesn't make local files system headers :)

We don't use `#include <variant.h>` anywhere in Godot, even though `core` is in the CPPPATH.

Comment (karroffel) [reply modules/gdnative/nativescript/nativescript.h]: @akien-mga at least all files inside of `modules/gdnative/include` should use it, as those are the files that are later used in the creation of libraries, where they are used *like* a system path.

Comment (akien-mga) [reply modules/gdnative/nativescript/nativescript.h]: Alright, you know best about the practical use case, I just wanted to point out the convention. Actual feature beats code style of course :)

**Pull request merged**